File: 669.修剪二叉搜索树.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    TreeNode* trimBST(TreeNode* root, int L, int R) {
        if(!root)   return NULL;
        if(root->val >= L && root->val <= R){
            root->left = trimBST(root->left, L, R);
            root->right = trimBST(root->right, L, R);
        } else if (root->val < L){
            root = trimBST(root->right, L, R);
        } else {
            root = trimBST(root->left, L, R);
        }
        return root;
    }
};

File: 二叉树/669.修剪二叉搜索树.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} low
 * @param {number} high
 * @return {TreeNode}
 */
var trimBST = function (root, low, high) {

    const preorder = (root, low, high) => {
        if (!root)
            return null;
        if (root.val >= low && root.val <= high) {
            root.left = preorder(root.left, low, high);
            root.right = preorder(root.right, low, high);
            return root;
        } else if (root.val < low) {
            return preorder(root.right, low, high);
        } else if (root.val > high) {
            return preorder(root.left, low, high);
        }
    }

    return preorder(root, low, high);
};